src/Agent.Sdk/Knob.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk
+{
+    public enum KnobSource
+    {
+        BuiltinDefault,
+        EnvironmentVariable,
+        RuntimeVariable,
+    };
+
+    public class KnobValue
+    {
+        public KnobSource Source { get;  private set;}
+        public string Which { get; private set; }
+        private string _value;
+        public KnobValue(string value, KnobSource source, string which=null)
+        {
+            _value = value;
+            Source = source;
+            Which = which;
+        }
+
+        public string AsString()
+        {
+            return _value;
+        }
+
+        public bool AsBoolean()
+        {
+            return StringUtil.ConvertToBoolean(_value);
+        }
+    }
+
+    public class AgentKnobs : KnobPanel
+    {
+        public static readonly IKnob UseNode10 = new Knob()
+        {
+            Name="UseNode10",
+            EnvironmentVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            RuntimeVariableNames=new List<string>{ "AGENT_USE_NODE10" },
+            Description="Forces the agent to use Node 10 handler for all Node-based tasks",
+            DefaultValue="false"
+        };
+    }
+
+    public class KnobPanel
+    {
+        public class Knob : IKnob
+        {
+            public string Name { get; set; }
+            public List<string> EnvironmentVariableNames { get; set; }
+            public List<string> RuntimeVariableNames { get; set; }
+            // public TBDType CommandLineOption {get; private set;}
+            public string Description { get; set; }
+            public bool IsDeprecated {get; set; } = false;  // is going away at a future date
```

Comment:
Nit: spacing.

Also, can we add more details to these comments? Do we know when it will go away/under what criteria?

Reply:
This may not be clear, but I was trying to say that IsDeprecated() returning true means that this Knob is going away at some point, not that the method is going away.

I am open to ideas about how we can best communicate to users of the agent that a Knob they are using has been deprecated and when support for it will be fully dropped.

Reply:
Mentioned below, I'd prefer a String to bool here